index.js

```diff
@@ -231,217 +231,197 @@ smartapp
-    let stateHistoryQuery = await redis.getStatesWindow(currentLocationId, '+', '-', 60);
-    let stateMemory = [];
-    //Store this in a local array
-    for(let i = 0; i < stateHistoryQuery.length; i++){
-      stateMemory.push(stateHistoryQuery[i].state)
-    }
-    // If RESET state is not succeeded by NO_PRESENCE_NO_SESSION, and already hasn't been artificially seeded, seed the sessions table with a reset state
-    for(let i=1; i<(stateHistoryQuery.length); i++){
-      if ( (stateHistoryQuery[i].state == STATE.RESET) && !( (stateHistoryQuery[i-1].state == STATE.NO_PRESENCE_NO_SESSION) || (stateHistoryQuery[i-1].state == STATE.RESET)) && !(resetDiscrepancies.includes(stateHistoryQuery[i].timestamp))){
-        console.log(`The Reset state logged at ${stateHistoryQuery[i].timestamp} has a discrepancy`);
-        resetDiscrepancies.push(stateHistoryQuery[i].timestamp);
-        console.log('Adding a reset state to the sessions table since there seems to be a discrepancy');
-        console.log(resetDiscrepancies);
-        await redis.addStateMachineData(STATE.RESET, currentLocationId);
-        //Once a reset state has been added, additionally reset any ongoing sessions
-        autoResetSession(currentLocationId);
-      }
+    // Iterating through multiple locations
+    for(let i = 0; i < locations.length; i++){
+        //Get recent state history
+        let currentLocationId = locations[i];
+        let stateHistoryQuery = await redis.getStatesWindow(currentLocationId, '+', '-', 60);
+        let stateMemory = [];
+        //Store this in a local array
+        for(let i = 0; i < stateHistoryQuery.length; i++){
+            stateMemory.push(stateHistoryQuery[i].state)
+        }
+        // If RESET state is not succeeded by NO_PRESENCE_NO_SESSION, and already hasn't been artificially seeded, seed the sessions table with a reset state
+        for(let i=1; i<(stateHistoryQuery.length); i++){
+            if ( (stateHistoryQuery[i].state == STATE.RESET) && !( (stateHistoryQuery[i-1].state == STATE.NO_PRESENCE_NO_SESSION) || (stateHistoryQuery[i-1].state == STATE.RESET)) && !(resetDiscrepancies.includes(stateHistoryQuery[i].timestamp))){
+                console.log(`The Reset state logged at ${stateHistoryQuery[i].timestamp} has a discrepancy`);
+                resetDiscrepancies.push(stateHistoryQuery[i].timestamp);
+                console.log('Adding a reset state to the sessions table since there seems to be a discrepancy');
+                console.log(resetDiscrepancies);
+                await redis.addStateMachineData(STATE.RESET, currentLocationId);
+                //Once a reset state has been added, additionally reset any ongoing sessions
+                autoResetSession(currentLocationId);
+            }
+        }
     }
-  }
 }, WATCHDOG_TIMER_FREQUENCY)
 
 // Handler for income SmartThings POST requests
-app.post('/api/st', function(req, res, next) {
+app.post('/api/st', function(req, res) {
```

Comment:
Confirming that our middleware doesn't use succession of any kind?

Reply:
@mariocimet

Reply:
Ooo, actually there is a neat way around this. We can add an `eslint-disable-next-line` comment right above it which will make it not complain about the unused `next` variable. We used this in Buttons for ones like this where it could useful in the future to remember that those parameters exist even if we don't use them now. I'll update this where I can.

Reply:
Fixed by adding `next` parameter back in and then telling `eslint` to not checkout for `no-unused-vars` on this line.